qemu/tests/block_hotplug.py

```diff
@@ -100,109 +113,70 @@ def unplug_device(vm, get_disk_cmd, device):
-                            controller_device_dict.setdefault(controller_id, []).append(device)
-                            break
-                    else:
-                        test.fail("Can't find device '%s' in qtree" % device.get_param("id"))
-            else:
-                if params.get("drive_format") in pci_type:
-                    get_disk_cmd += " | egrep -v '^/dev/[hsv]da[0-9]*$'"
-
-                device.set_param("id", img_list[num + 1])
-                new_disks = find_disk(vm, get_disk_cmd)
 
-            device_list.append(device)
-            if not new_disks:
-                test.fail("Cannot find new disk after hotplug.")
-
-            if params.get("need_plug") == "yes":
-                disk = new_disks[0]
-            else:
-                disk = new_disks[num]
-
-            session = vm.wait_for_login(timeout=timeout)
-            if params.get("os_type") == "windows":
-                if iteration == 0:
-                    error_context.context("Format disk", logging.info)
-                    utils_misc.format_windows_disk(session, disk_index[num],
-                                                   mountpoint=disk_letter[num])
-            error_context.context("Check block device after hotplug.",
-                                  logging.info)
-            if params.get("disk_op_cmd"):
-                if params.get("os_type") == "linux":
-                    test_cmd = params.get("disk_op_cmd") % (disk, disk)
-                elif params.get("os_type") == "windows":
-                    test_cmd = params.get("disk_op_cmd") % (disk_letter[num],
-                                                            disk_letter[num])
-                    test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
+                session = vm.wait_for_login(timeout=timeout)
+                if params.get("os_type") == "windows":
+                    if iteration == 0:
+                        error_context.context("Format disk", logging.info)
+                        utils_misc.format_windows_disk(session, disk_index[num],
+                                                       mountpoint=disk_letter[num])
+                error_context.context("Check block device after hotplug.",
+                                      logging.info)
+                if params.get("disk_op_cmd"):
+                    if params.get("os_type") == "linux":
+                        test_cmd = params.get("disk_op_cmd") % (disk, disk)
+                    elif params.get("os_type") == "windows":
+                        test_cmd = params.get("disk_op_cmd") % (disk_letter[num],
+                                                                disk_letter[num])
+                        test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
+                    else:
+                        test.cancel("Unsupported OS type '%s'" % params.get("os_type"))
+
+                    status, output = session.cmd_status_output(test_cmd,
```

Comment:
output is not used, either use it in error message on line 163 or use `cmd_status` instead.